demo/site/src/news/NewsPage.tsx

```diff
@@ -0,0 +1,79 @@
+"use client";
+import { DamImageBlock } from "@src/common/blocks/DamImageBlock";
+import { fetcher } from "@src/util/Fetcher";
+import Link from "next/link";
+import { useState } from "react";
+import styled from "styled-components";
+import useSWR from "swr";
+
+import { GQLNewsIndexPageQuery } from "./NewsPage.loader.generated";
+
+type News = GQLNewsIndexPageQuery["newsList"];
+
+export function NewsPage({ initialData, limit }: { initialData: News; limit: number }) {
+    const [pages, setPages] = useState<number[]>([]); // Contains array of offsets
+
+    return (
+        <>
+            <h1>News</h1>
+            <CardList>
+                <NewsList news={initialData.nodes} />
+                {pages.map((offset) => (
+                    <NewsListPaged key={offset} offset={offset} limit={limit} />
+                ))}
+            </CardList>
+            <div>
+                {initialData.totalCount > (pages.at(-1) || 0) * limit && (
+                    <button
+                        onClick={async () => {
+                            setPages([...pages, (pages.length + 1) * limit]);
+                        }}
+                    >
+                        Load more
+                    </button>
+                )}
+            </div>
+        </>
+    );
+}
+
+function NewsListPaged({ offset, limit }: { offset: number; limit: number }) {
+    const { data, isLoading, error } = useSWR<News>({ url: "news/api", params: { offset, limit } }, { fetcher });
```

Comment:
@nsams recently argued that using relative URLs isn't a good idea, so should we use an absolute URL here?

Reply:
In a Component that can be rendered anywhere (such a s a Link Block) it must be absolute. In this situation we are in a page component and that can't be rendered anywhere, so it works. I still would use absolute urls everywhere, mainly for consistency.

Reply:
In combination with predefined pages (where /news could become /aktuelles) it probably makes sense to use the current path

Reply:
I thought about hardcoding the full absolute path, to have one way of creating paths - that works everywhere.

We really should add predefined pages to demo!

Reply:
We have predefined pages in demo.

Reply:
oh, good!

but that means `<Card key={news.id} href={`news/${news.slug}`}>` is wrong?